src/main/java/org/opensearch/flowframework/model/Template.java

```diff
@@ -132,6 +144,18 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         }
         xContentBuilder.endObject();
 
+        xContentBuilder.startObject(RESPONSES_FIELD);
+        for (Entry<String, Object> e : responses.entrySet()) {
+            xContentBuilder.field(e.getKey(), e.getValue());
+        }
```

Comment:
I think it's a good idea to use generic type when try to prevent unknown value during parsing stage. But the responses/user_outputs map here has the exact same data structure as user_inputs. So I'm just keeping it the same as what user_inputs field is using. I will document this in a global context followup issue future reference.  Will attache this pr to the issue later